platform/engineering/frontend/eslint/eslint-testing.md

```diff
@@ -0,0 +1,57 @@
+# ESLint Rule Changes and Testing
+
+**Author:** Jhonny Gonzalez
+**Last Updated:** February 21, 2020
+**eslint version:** 4.18.2
+
+## Overview
+
+This is an initial process to test and deploy new rules for ESLint.
+
+## Objective
+
+To improve the code coverage and code quality in the repo by enforcing new ESLint rules. This also will help all VFS teams to write better code.
+
+## Background
+
+We installed ESLint a couple of years ago in the VA repo and hasn't been updated nor improved since then.
+
+Now, that VFS teams will have an increased ownership on their pull request reviews, this will help them getting a better judgement as well as facilitate the review process.
+
+# Specifics
+
+## ESLintin Procedure
+
+The ESLinting rules in CircleCI will be used as an experimental staging area as we explore new rules and ways to communicate any issues that are discovered.
+
+## Code Location
+
+### ESLint main file
+
+`.eslintrc` => located at the root folder.
+
+### ESLint Testing file
+
+`circle.eslint.json` => rules under review will be located here
+
+## Plugin/New Rules Audit
+
+Before contemplating the idea of adding a new plugin with rules, it is important to perform an audit. This audit will help us determine if the current repo is in compliance with each rule. All outstanding errors from rules must be cleared (disabled or fixed) before going into the testing stage.
```

Comment:
```suggestion
Before adding a new plugin or new rules, it is important to perform an audit. This audit will help us determine if the current repo is in compliance with each rule. All outstanding errors from rules must be cleared (disabled or fixed) before going into the testing stage.
```
So the important part here are the _rules_ we're running, not just the plugins. We can disable rules on new plugins, so I think it's important to treat each rule individually.

> This audit will help us determine if the current repo is in compliance with each rule.

Are we trying to determine if we're in compliance with each _new_ rule, or each _existing_ rule?

> All outstanding errors from rules must be cleared (disabled or fixed) before going into the testing stage.

Same question: Are these errors from the new rules we want to enforce?

Reply:
Got it. Let me phrase it better.